include/circt/Dialect/FIRRTL/OpExpressions.td

```diff
@@ -403,3 +403,15 @@ def StdIntCast : FIRRTLOp<"stdIntCast", [NoSideEffect]> {
 
   let verifier = [{ return ::verifyStdIntCast(*this); }];
 }
+
+// This operation converts from an flipped SInt/UInt to a signless integer type.
+def StdFlippedIntCast : FIRRTLOp<"stdFlippedIntCast", [NoSideEffect]> {
+  let arguments = (ins AnyType:$input);
+  let results = (outs AnyType:$result);
+
+  let assemblyFormat = [{
+    $input attr-dict `:` functional-type($input, $result)
+  }];
+
+  let verifier = [{ return ::verifyStdFlippedIntCast(*this); }];
+}
```

Comment:
I'm not sure why you need a cast.  Is this just to facilitate lowering and the cast should disappear when lowering is complete?

Reply:
The cast is to handle casting from !firrlt.flip<uint<N>> to iN. It's sort of modeled after the stdIntCast which handles lowering `firrtl.asUInt %in1 : (!firrtl.uint<4>) -> !firrtl.uint<4>` to `firrtl.stdIntCast %in1 : (!firrtl.uint<4>) -> i4`.

The use case here is that rtl.connect requires each argument to have the same type, and as far as I can tell, there's no support for casting a flipped type to its passive type. That said, I think bundles and vectors can be flipped, so this cast is incomplete.

What would you suggest instead of this cast?

Reply:
It's not clear to me what the relationship between the RTL and FIRRTL dialects is.  it seems that firrtl is being used for composing blocks, but the RTL dialect is being used inside blocks?  I think this could go two ways:
1) completely lower from FIRRTL to RTL.
2) use hiearchy so that the rtl.module operation handles the type conversion at the boundaries, presenting FIRRTL types externally and std types internally.

Reply:
I don't think rtl.module exists yet, but I like option 2.

Reply:
Yeah the goal should be to achieve #1 (completely lower form firrtl to RTL) but we're not there yet - this is a work in progress.

Amalee, did you consider extending `firrtl.stdIntCast` to allow/strip flip types in the firrtl input/result?

Reply:
Yes, I was initially considering that there are other types that may need to be flipped, but for now it's maybe cleaner to have it combined.